src/lib/logical_query_plan/delete_node.cpp

```diff
@@ -29,4 +33,11 @@ bool DeleteNode::subtree_is_read_only() const { return false; }
 
 const std::string& DeleteNode::table_name() const { return _table_name; }
 
+bool DeleteNode::shallow_equals(const AbstractLQPNode& rhs) const {
+  Assert(rhs.type() == type(), "Can only compare nodes of the same type()");
```

Comment:
Suggestions: Could we move this check in down in a non-virtual method that calls `_on_shallow_equals` or so?

Reply:
yeah, considered that but don't like additional layers of function dispatches for avoiding some trivial lines. Also, countless people would look at `_on_shallow_equals` and wonder how it makes sure the types match. Let's spare them the reading.